### DISCUSSION TITLE
new_tab / new_split with command

### DISCUSSION THREAD

**Discussion by cbrst** [ghostty-org/ghostty#3379] [Feature Requests, Ideas] [answered]
It would be neat to have the option to run a specific command with the `new_tab` and `new_split` actions.

For example, in wezterm I have a keybinding that opens a split with lazygit running, since I prefer to have it run "natively" in a terminal, instead of the nvim integrated terminal. I can quickly do what needs to be done and hit "q" to quit lazygit, close the split and return to nvim.

Having the same functionality in Ghostty instead of having to open a split, manually launch lazygit, close it and manually close the split would be neat.

**Comment by jvortmann** [comment]
There is a plugin for tmux that does a similar thing and can be used as inspiration for the implementation. This is it [tmux-xpanes](https://github.com/greymd/tmux-xpanes). It does for splits, but I think the general idea of "opening a command somewhere else" is the thing here. I would be really nice to make it possible to run something and indicate where (tab, split, window, etc)

**Reply by cbrst** [reply]
xpanes seems really complex. I just want `tmux split-window -h "command"`, just builtin to the terminal emulator, without adding another layer of abstraction. 😅

**Comment by mitchellh** [comment, marked as answer]
Depends on #2353
